### PR TITLE
feat: collection index endpoint

### PR DESCRIPTION
**Pull request by ebezzi**
### Reviewers
**Functional:** 
@Bento007 @MillenniumFalconMechanic @NoopDog 

**Readability:** 

Ticket [#1632](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1632)

---

## Changes
- Add a `/collections/index` endpoint. `GET`, no parameters
- Returns an array with all the public, non tombstoned collections and a subset of the fields.

### CODE REVIEW

**Review by MillenniumFalconMechanic — APPROVED** [review]

File: tests/unit/backend/corpora/api_server/test_v1_collection.py [hunk 1]
```diff
@@ -603,6 +603,41 @@ def test__list_collection__check_owner(self):
             self.assertIn(private_owned, ids)
             self.assertNotIn(private_not_owned, ids)
 
+    def test__get_all_collections_for_index(self):
+        # dataset = self.generate_dataset(self.session, id="test_dataset_id_for_index", cell_count=42)
```
Comment (Bento007) [inline tests/unit/backend/corpora/api_server/test_v1_collection.py:607]: remove commented out code

File: tests/unit/backend/corpora/api_server/test_v1_collection.py [hunk 2]
```diff
@@ -603,6 +603,41 @@ def test__list_collection__check_owner(self):
...
+        collection_id_private = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="test_user_id"
+        ).id
+
+        collection_id_tombstone = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PUBLIC.name, tombstone=True, owner="test_user_id"
+        ).id
+
+        test_url = furl(path="/dp/v1/collections/index")
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
```
Comment (Bento007) [inline tests/unit/backend/corpora/api_server/test_v1_collection.py:624]: Is an auth token need for this endpoint?

**Review by Bento007 — CHANGES_REQUESTED** [review]

File: tests/unit/backend/corpora/api_server/test_v1_collection.py (hunk 2 above)
Comment (ebezzi) [reply tests/unit/backend/corpora/api_server/test_v1_collection.py]: Nope, good call. Thanks!

File: backend/config/corpora-api.yml [hunk 3]
```diff
@@ -413,6 +413,39 @@ paths:
         "404":
           $ref: "#/components/responses/404"
 
+  /dp/v1/collections/index:
```
Comment (Bento007) [inline backend/config/corpora-api.yml:416]: I think `/dp/v1/collections:` could be modified to support this need. Why is a new endpoint needed?

File: backend/config/corpora-api.yml [hunk 4]
```diff
@@ -413,6 +413,39 @@ paths:
...
+    get:
+      tags:
+        - collections
+      summary: >-
+        Returns all public, non tombstoned collections. This endpoint is meant to be used for displaying collections
+        in a compact view. It doesn't return all the fields, so it is not suitable for general usage.
+      security:
```
Comment (Bento007) [inline backend/config/corpora-api.yml:423]: This can be removed if the functionality is the name for both authenticated and unauthenticated.

**Review by Bento007 — COMMENTED** [review]
What do we need a new endpoint for this?

File: backend/config/corpora-api.yml (hunk 3 above)
Comment (ebezzi) [reply backend/config/corpora-api.yml:416]: If we want to reuse the existing endpoint, we'd have to add parameters to specify whether to only return public and not tombstoned collections, plus add new fields (collection name and a few dates). I am not opposed to it but I think it's faster/cleaner to create a new one. Also, @MillenniumFalconMechanic , once we have the new views running, will the existing endpoint be needed at all?

**Review by Bento007 — APPROVED** [review]

File: backend/config/corpora-api.yml (hunk 3 above)
Comment (MillenniumFalconMechanic) [reply backend/config/corpora-api.yml:416]: The existing `/dp/v1/collections` endpoint will still be required for the my collections functionality.

**Pull request merged**